### PR TITLE
PMM-10248 Fix lint issues in install_tarball

### PR DESCRIPTION
**Pull request by cezmunsta**
[PMM-10248](https://jira.percona.com/browse/PMM-10248)

Rebase after #2613 is merged

### CODE REVIEW

File: build/bin/install_tarball [hunk 1]
```diff
@@ -2,59 +2,63 @@
...
-for FILE in $( ls ${PWD}/bin ); do
-    if [ "x${FILE}" = "xpmm-admin" -o "x${FILE}" = "xpmm-agent" ]; then
-        ${INSTALL_COMMAND} -m 0755 ${PWD}/bin/${FILE} ${PMM_DIR}/bin
-    elif [ "x${FILE}" = "xpt-summary" -o "x${FILE}" = "xpt-mysql-summary" -o "x${FILE}" = "xpt-pg-summary" -o "x${FILE}" = "xpt-mongodb-summary" ]; then
-        ${INSTALL_COMMAND} -m 0755 ${PWD}/bin/${FILE} ${PMM_DIR}/tools
+find ./bin -maxdepth 1 -type f | while read -r FILE; do
+    if [ "x${FILE}" = "xpmm-admin" ] || [ "x${FILE}" = "xpmm-agent" ]; then
+        ${INSTALL_COMMAND} -m 0755 "${FILE}" "${PMM_DIR}"/bin
+    elif [ "x${FILE}" = "xpt-summary" ] || [ "x${FILE}" = "xpt-mysql-summary" ] || [ "x${FILE}" = "xpt-pg-summary" ] || [ "x${FILE}" = "xpt-mongodb-summary" ]; then
+        ${INSTALL_COMMAND} -m 0755 "${FILE}" "${PMM_DIR}"/tools
```
Comment (nikita-b) [inline build/bin/install_tarball:49]: To be honest, readability is worse than was.

In my opinion, we should write scripts as clearly as possible and this should be a priority.

Let's change it to:
```
        ${INSTALL_COMMAND} -m 0755 "${FILE}" "${PMM_DIR}/tools"
```
Here and in other places.

**Review by nikita-b — CHANGES_REQUESTED** [review]

File: build/bin/install_tarball (hunk 1 above)
Comment (cezmunsta) [reply build/bin/install_tarball:49]: @nikita-b 

```bash
[ "x${FILE}" = "xpmm-admin" ] || [ "x${FILE}" = "xpmm-agent" ]
```

I would argue that boolean operations are more readable as the syntax is commonly present in other languages, plus it is better defined in shells.

Similarly,

```text
"${PMM_DIR}/tools"
```

.. this is less readable, especially where there is no syntax-highlighting, to pick out the variable as you then need to read backwards from the `"`, which draws the eye to the end. The prevention of globbing is specific to the user-defined values, so where does that leave the other parts of the script where it is not necessary?

For example:
```bash
${RESOLUTION}-resolution
${INSTALL_COMMAND} 
```

These are both well-defined and do not risk expansion, so I would argue that something such as `"${PMM_DIR}"/tools` makes things clear.

Comment (cezmunsta) [reply build/bin/install_tarball:49]: Just to clarify the part about other shells and lack of syntax-highlighting, you may be on a server where you do not have admin permissions to install packages, `bash` is not present everywhere and `less`, `more`, or `vi` may be all that you have to read the file before executing it.

**Review by EvgeniyPatlan — APPROVED** [review]

File: build/bin/install_tarball (hunk 1 above)
Comment (EvgeniyPatlan) [reply build/bin/install_tarball:49]: totally agree with @cezmunsta :+1:

**Review by EvgeniyPatlan — APPROVED** [review]

File: build/bin/install_tarball (hunk 1 above)
Comment (nikita-b) [reply build/bin/install_tarball:49]: @cezmunsta 
I'm not sure we should sacrifice the convenience of all in favor people don't have syntax-highlighting. Yes, sometimes people don't have syntax-highlighting but it's rare these days.

The main users of our script have highlighting and they should be our `target auditory`. If we can help others, that's great but without decrease to the comfort of others.

In any case, I think it's better to follow the guides and I don't see any strong recommendation about this:
https://google.github.io/styleguide/shellguide.html#quoting

You can choose any way that think will be better.

**Review by nikita-b — APPROVED** [review]

File: build/bin/install_tarball (hunk 1 above)
Comment (cezmunsta) [reply build/bin/install_tarball:49]: > The main users of our script have highlighting

Can you qualify that with data? :)

Comment (nikita-b) [reply build/bin/install_tarball:49]: @cezmunsta if most administrators and developers don't even have access to `vim`, then I think IT industry is failing :)

Comment (cezmunsta) [reply build/bin/install_tarball:49]: That is not data

**Comment by denisok** [comment]
build fails:
```
Getting https://github.com/percona/pmm-managed/archive/23069cdf1bb886cb2b82b9814ecb62ff67a362b8/pmm-managed-23069cd.tar.gz to SOURCES/pmm-managed-23069cd.tar.gz

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                 Dload  Upload   Total   Spent    Left  Speed


  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0


  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

curl: (22) The requested URL returned error: 404 Not Found

```

so `pmm-managed` was moved to `pmm` repo, you probably need to re-sync submodules?

File: build/bin/install_tarball (hunk 1 above)
Comment (cezmunsta) [reply build/bin/install_tarball:49]: @EvgeniyPatlan any thoughts?

**Comment by cezmunsta** [comment]
> so `pmm-managed` was moved to `pmm` repo, you probably need to re-sync submodules?

@denisok this is a PR _for_ `pmm-submodules`

**Comment by denisok** [comment]
@cezmunsta I know. But you have all submodules here, so you need to have build passed. @nikita-b could you please help with the build?

File: build/bin/install_tarball (hunk 1 above)
Comment (nikita-b) [reply build/bin/install_tarball:49]: @cezmunsta it works for both sides. Do you have accurate data about PMM customers?

Accurate data is telemetry or at least public surveys(much much worse).

We can play with data endlessly, but in my opinion, the assumption that even if there are people without `vim` on the servers, they can edit it on the server with `vim`/`vscode` is quite strong.

If we can help these people and don't disturb others, it's great. But if we'll decrease readability for it...It looks very strange to me.

**Comment by cezmunsta** [comment]
@denisok is `install_tarball` used in the tests?

**Comment by denisok** [comment]
@cezmunsta it is not a test, it is a build. Yes, it is used there. But I don't think it is a root cause.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-2614-bb0bef7
client docker - perconalab/pmm-client-fb:PR-2614-bb0bef7
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-2614-bb0bef7.tar.gz
Create Staging Instance: https://pmm.cd.percona.com/job/aws-staging-start/parambuild/?DOCKER_VERSION=perconalab/pmm-server-fb:PR-2614-bb0bef7&CLIENT_VERSION=https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-2614-bb0bef7.tar.gz

**Comment by JNKPercona** [comment]
API tests have failed, Please check: API: https://pmm.cd.percona.com/job/pmm2-api-tests/10577/

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-2614-ab24ed8
client docker - perconalab/pmm-client-fb:PR-2614-ab24ed8
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-2614-ab24ed8.tar.gz
Create Staging Instance: https://pmm.cd.percona.com/job/aws-staging-start/parambuild/?DOCKER_VERSION=perconalab/pmm-server-fb:PR-2614-ab24ed8&CLIENT_VERSION=https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-2614-ab24ed8.tar.gz

**Comment by JNKPercona** [comment]
API tests have failed, Please check: API: https://pmm.cd.percona.com/job/pmm2-api-tests/10668/

**Pull request merged**